src/MongoDB.Driver/Linq/Linq3Implementation/Serializers/KnownSerializers/KnownSerializerFinder.cs

```diff
@@ -22,33 +27,139 @@ internal class KnownSerializerFinder : ExpressionVisitor
+        private KnownSerializerFinder(Expression root, IBsonDocumentSerializer rootSerializer)
         {
+            _rootSerializer = rootSerializer;
+            _root = root;
         }
 
         // public methods
         public override Expression Visit(Expression node)
         {
-            _expressionKnownSerializers = new KnownSerializersNode(_expressionKnownSerializers);
-            _registry.Add(node, _expressionKnownSerializers);
+            if (node == null) return null;
+
+            _currentKnownSerializersNode = new KnownSerializersNode(_currentKnownSerializersNode);
+
+            if (node == _root)
+            {
+                _currentSerializer = _rootSerializer;
+            }
 
             var result = base.Visit(node);
+            _registry.Add(node, _currentKnownSerializersNode);
+            _currentKnownSerializersNode = _currentKnownSerializersNode.Parent;
+            return result;
+        }
+
+        protected override Expression VisitMember(MemberExpression node)
+        {
+            var result = base.VisitMember(node);
+            if (_currentSerializer.TryGetMemberSerializationInfo(node.Member.Name, out var memberSerializer))
+            {
+                PropagateToRoot(node, memberSerializer.Serializer);
+
+                if (memberSerializer.Serializer is IBsonDocumentSerializer bsonDocumentSerializer)
+                {
+                    _currentSerializer = bsonDocumentSerializer;
+                }
+            }
+            return result;
+        }
+
+        protected override Expression VisitMethodCall(MethodCallExpression node)
+        {
+            var result = base.VisitMethodCall(node);
+
+            if (node.Method.Is(QueryableMethod.OfType) || node.Method.Is(EnumerableMethod.OfType))
+            {
+                var actualType = node.Method.GetGenericArguments()[0];
+                var serializer = BsonSerializer.LookupSerializer(actualType);
+                PropagateToRoot(node, serializer);
+            }
 
-            _expressionKnownSerializers = _expressionKnownSerializers.Parent;
             return result;
         }
+
+        protected override Expression VisitNew(NewExpression node)
+        {
+            var result = base.VisitNew(node);
+
+            if (node.Type == _rootSerializer.ValueType)
+            {
+                return result;
+            }
+
+            IBsonSerializer serializer;
+            if (node.Type == typeof(DateTime))
+            {
+                serializer = new DateTimeSerializer();
+            }
+            else if (node.Type == typeof(DateTimeOffset))
+            {
+                serializer = new DateTimeOffsetSerializer();
+            }
+            else
+            {
+                var classMapType = typeof(BsonClassMap<>).MakeGenericType(node.Type);
+                var classMap = (BsonClassMap)Activator.CreateInstance(classMapType);
+                classMap.AutoMap();
```

Comment:
By calling `AutoMap` we're not using any known serializers for the child properties.

Reply:
We're going to defer this issue.